### PR TITLE
Created underlying token symbols next to the token address on the das…

### PR DESCRIPTION
**Pull request by josephchimebuka**
Fixs #188

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -20,6 +21,8 @@
     transform_loans_data,
     transform_main_chart_data,
 )
+from typing import Dict, Set
```
Comment (djeck1432) [inline app.py:24]: Please, run `isort` command https://pycqa.github.io/isort/ for this file
p.s. this package is already install, just run command to format `imports`

File: app.py [hunk 2]
```diff
@@ -28,6 +31,36 @@
 ]  # "Hashstack V0", "Hashstack V1"
 
 
+
+    
+def fetch_token_symbols2(token_addresses: Set[str]) -> Dict[str, str]:
```
Comment (djeck1432) [inline app.py:36]: add docstring please

Comment (djeck1432) [reply app.py:36]: Why do you have `fetch_token_symbols2` 
Let's come up with better naming

File: app.py [hunk 3]
```diff
@@ -28,6 +31,36 @@
...
+    async def async_fetch():
+        return {addr: await get_underlying_token_symbol(addr) for addr in token_addresses}
+    return asyncio.run(async_fetch())
+
+async def get_underlying_token_symbol(token_address: str) -> str:
```
Comment (djeck1432) [inline app.py:41]: I think it should be in `chart_utils.py` not here
Also add docstring

File: app.py [hunk 4]
```diff
@@ -28,6 +31,36 @@
...
+            calldata=[],
+        )
+        underlying_token_address = src.helpers.add_leading_zeros(
+            hex(underlying_token_address[0])
+        )
+        underlying_token_symbol = await src.helpers.get_symbol(
+            token_address=underlying_token_address
+        )
+        return underlying_token_symbol.strip('\x00')
+    except Exception:
```
Comment (djeck1432) [inline app.py:55]: Be more specific in `Exception` which you are trying to catch

File: app.py [hunk 5]
```diff
@@ -28,6 +31,36 @@
...
+            selector="underlyingAsset",
+            calldata=[],
+        )
+        underlying_token_address = src.helpers.add_leading_zeros(
+            hex(underlying_token_address[0])
+        )
+        underlying_token_symbol = await src.helpers.get_symbol(
+            token_address=underlying_token_address
+        )
+        return underlying_token_symbol.strip('\x00')
```
Comment (djeck1432) [inline app.py:54]: Please, move it to `constant` variable `'\x00'`
In python it's call magic string read more here: https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings

File: app.py [hunk 6]
```diff
@@ -169,6 +202,27 @@ def process_liquidity(
 
     return main_chart_data, collateral_token_price
 
+def extract_token_addresses(loans_data: pandas.DataFrame) -> Set[str]:
```
Comment (djeck1432) [inline app.py:205]: add docstring

File: app.py [hunk 7]
```diff
@@ -169,6 +202,27 @@ def process_liquidity(
...
+    
+    for _, row in loans_data.iterrows():
+        if isinstance(row['Collateral'], dict):
+            collateral_addresses.update(row['Collateral'].keys())
+        if isinstance(row['Debt'], dict):
+            debt_addresses.update(row['Debt'].keys())
+    
+    return collateral_addresses.union(debt_addresses)
+
+def update_loan_data_with_symbols(loans_data: pandas.DataFrame, token_symbols: Dict[str, str]) -> pandas.DataFrame:
```
Comment (djeck1432) [inline app.py:217]: add docstring

File: app.py [hunk 8]
```diff
@@ -169,6 +202,27 @@ def process_liquidity(
...
+    def update_column(col):
+        if isinstance(col, dict) and col:  # Check if it's a non-empty dictionary
+            return {f"{addr} ({token_symbols.get(addr, 'Unknown')})": val for addr, val in col.items()}
+        return col  # Return the original value if it's not a non-empty dictionary
+    
+    loans_data['Collateral'] = loans_data['Collateral'].apply(update_column)
```
Comment (djeck1432) [inline app.py:223]: Please, run `black` command to have proper formattting

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by josephchimebuka** [comment]
Hello @djeck1432 and @lukaspetrasek pls can you help me review my code I've made the requested changes

**Comment by djeck1432** [comment]
While testing your PR, I had next error, please fix it:

```
starknet_py.net.client_errors.ClientError: Client failed with code 40. Message: Contract error. Data: {'revert_error': 'Entry point EntryPointSelector(0x12d60825cc57ee4a44efdb81f0050fb99bcc550c411acb3ca38c9b83103dc2f) not found in contract.'}

```
in `fetch_token_symbols_from_set_of_loan_addresses`

@josephchimebuka

**Comment by josephchimebuka** [comment]
Hello @djeck1432 I have fix changes and it is running on local as well

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by josephchimebuka** [comment]
Used isort should be better now @djeck1432

File: app.py [hunk 9]
```diff
@@ -15,16 +15,27 @@
...
+    fetch_token_symbols_from_set_of_loan_addresses,
+    update_loan_data_with_symbols,
+)
+
+PROTOCOL_NAMES = [
+    "zkLend",
+    "Nostra Alpha",
+    "Nostra Mainnet",
+]  # "Hashstack V0", "Hashstack V1"
+
```
Comment (djeck1432) [inline app.py:34]: remove blank lines
better will be if you use `black` command for this file

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Please, sync with master branch

**Review by djeck1432 — APPROVED** [review]

**Review by djeck1432 — APPROVED** [review]
Looks godo to me

**Pull request merged**